### PR TITLE
Pypy bridge foreign exn tests

### PR DESCRIPTION
**Pull request by vext01**
This and https://bitbucket.org/softdevteam/pypy-hippy-bridge/pull-requests/3/fix-for-throwing-foreign-exceptions/diff fixes a bug and tests more exception paths with regards to foreign exceptions. The fix itself is in the PyPy change.

Sorry for the delay, it took several attempts to get right, and it turns out the fix is a one liner.

Looks OK?

*Please don't merge yet, as this has not yet been benchmarked. I will report back with benchmark outcomes soon*

### CODE REVIEW

File: testing/test_pypy_bridge_scope.py [hunk 1]
```diff
@@ -1061,3 +1061,128 @@ def f():
...
+                return "fail"
+            except PHPException as e:
+                return e.message
+        EOD;
+        $f = compile_py_func($pysrc);
+        echo $f();
+        ''')
+        assert php_space.str_w(output[0]) == "most illogical"
+
+    # XXX: We yet catch a foreign PHP exception under Python by name.
```
Comment (ltratt) [inline testing/test_pypy_bridge_scope.py]: I can't parse this comment.

File: testing/test_pypy_bridge_scope.py [hunk 2]
```diff
@@ -1061,3 +1061,128 @@ def f():
...
+    @pytest.mark.xfail
+    def test_raise_foregin_exception007(self, php_space):
+        output = self.run(r'''
+        $pysrc = <<<EOD
+        def f():
+            def g():
+                raise LogicException("most illogical")
```
Comment (ltratt) [inline testing/test_pypy_bridge_scope.py]: `LogicException` isn't defined anywhere?

**Comment by vext01** [comment]
I forgot to explain the xfailing tests. This bug describes the issue: https://bitbucket.org/softdevteam/pypy-hippy-bridge/issues/33/catching-foreign-exceptions-by-name

I don't consider this high priority right now. Would you agree?

File: testing/test_pypy_bridge_scope.py (hunk 1 above)
Comment (vext01) [inline testing/test_pypy_bridge_scope.py]: bah. "We *can't* yet". WIll fix.

File: testing/test_pypy_bridge_scope.py (hunk 2 above)
Comment (vext01) [inline testing/test_pypy_bridge_scope.py]: It's a PHP builtin exception.

Comment (ltratt) [inline testing/test_pypy_bridge_scope.py]: Aha! OK.

**Comment by vext01** [comment]
OK, this is now benchmarked. No noticeable performance change. Merge?

*Don't forget to merge on the other repo referenced above too*.

**Pull request merged**